.github/workflows/reusable-golangci-lint.yml

```diff
@@ -72,7 +77,18 @@ jobs:
         working-directory: ${{ inputs.WORKING_DIRECTORY }}
         run: |
           grep replace go.mod || true
-          go mod tidy     
+          go mod tidy
+      # Check if graphql directory exists
+      - name: Check graphql dir existence
+        id: check_graph_dir
+        working-directory: ${{ inputs.WORKING_DIRECTORY }}
+        run: |
+          [ ! -z ${{ inputs.GQL_DIRECTORY }} ] && [ -d ${{ inputs.GQL_DIRECTORY }} ] && echo "dir_exists=true" >> "$GITHUB_OUTPUT"
+      # Run gqlgen to generate the generated.go file if check_graph_dir step is true
+      - name: Run gqlgen
+        working-directory: ${{ inputs.WORKING_DIRECTORY }}
+        if: steps.check_graph_dir.outputs.dir_exists == 'true'
+        run: make gen
```

Comment:
rename the make sub command to `make gqlgen` because `gen` is too general and we might use it for other purpose in the future

Reply:
Done, sorry I accidentally used force push. Also update README for the new `GQL_DIRECTORY` input parameter.